src/app/calculator/furnaces/opening/opening-form/opening-form.component.ts

```diff
@@ -0,0 +1,259 @@
+    this.openingService.updateDataArray(currentOpeningLoss, this.index, this.isBaseline);
+  }
+
+  removeLoss() {
+    this.openingService.removeLoss(this.index);
+  }
+
+  getArea() {
+    let smallUnit = 'in';
+    let largeUnit = 'ft';
+    if (this.settings.unitsOfMeasure === 'Metric') {
+      smallUnit = 'mm';
+      largeUnit = 'm';
+    }
+    if (this.openingLossesForm.controls.numberOfOpenings.valid && this.openingLossesForm.controls.lengthOfOpening.valid  && this.openingLossesForm.controls.heightOfOpening.valid) {
+      this.totalArea = 0.0;
+      if (this.openingLossesForm.controls.openingType.value === 'Round') {
+          this.openingLossesForm.controls.heightOfOpening.setValue(0);
+          //let radiusFeet = (radiusInches * .08333333) / 2;
+          let radiusInches = this.openingLossesForm.controls.lengthOfOpening.value;
+          let radiusFeet = this.convertUnitsService.value(radiusInches).from(smallUnit).to(largeUnit) / 2;
+          this.totalArea = Math.PI * Math.pow(radiusFeet, 2) * this.openingLossesForm.controls.numberOfOpenings.value;
+      } else if (this.openingLossesForm.controls.openingType.value === 'Rectangular (or Square)') {
+          let lengthFeet = 0;
+          let heightFeet = 0;
+          if (this.openingLossesForm.controls.lengthOfOpening.value) {
+            lengthFeet = this.convertUnitsService.value(this.openingLossesForm.controls.lengthOfOpening.value).from(smallUnit).to(largeUnit);
+          }
+          if (this.openingLossesForm.controls.heightOfOpening.value) {
+            heightFeet = this.convertUnitsService.value(this.openingLossesForm.controls.heightOfOpening.value).from(smallUnit).to(largeUnit);
+          }
+          this.totalArea = lengthFeet * heightFeet * this.openingLossesForm.controls.numberOfOpenings.value;
+      } 
+    }
+  }
+
+  roundVal(val: number, digits: number) {
+    let rounded = Number(val.toFixed(digits));
+    return rounded;
+  }
+
+  initFlueGasModal() {
+    this.showFlueGasModal = true;
+    this.openingService.modalOpen.next(this.showFlueGasModal);
+    this.flueGasModal.show();
+  }
+
+  hideFlueGasModal(calculatedAvailableHeat?: any) {
+    if (calculatedAvailableHeat) {
+      calculatedAvailableHeat = this.roundVal(calculatedAvailableHeat, 1);
+      this.openingLossesForm.patchValue({
+        availableHeat: calculatedAvailableHeat
+      });
+    }
+    this.flueGasModal.hide();
+    this.showFlueGasModal = false;
+    this.openingService.modalOpen.next(this.showFlueGasModal);
+  }
+
+  closeOperatingHoursModal() {
+    this.showOperatingHoursModal = false;
+  }
+
+  openOperatingHoursModal() {
+    this.showOperatingHoursModal = true;
+  }
+
+  updateOperatingHours(oppHours: OperatingHours) {
+    this.openingService.operatingHours = oppHours;
+    this.openingLossesForm.controls.hoursPerYear.patchValue(oppHours.hoursPerYear);
+    this.calculate();
+    this.closeOperatingHoursModal();
+  }
+
+  setOpHoursModalWidth() {
```

Comment:
This needs to be called ngAfterViewInit and on window resize (same as charge materials)

`if(this.formElement)`